app/models/concerns/fuzzy_search.rb

```diff
@@ -0,0 +1,69 @@
+module FuzzySearch
+  extend ActiveSupport::Concern
+
+  class_methods do
+    attr_reader :search_joins
+
+    ##
+    # Fuzzy search entry point
+    #
+    # @todo \
+    #   - Support join table properties (needed for other models)\
+    #   - Potential speed improvements (trigram indexes, using <-> operators)\
+    #   - Assess whether order_by has priority issues
+    #
+    # @param [string] search_text the text to search for
+    # @return [ActiveRecord::Relation]
+    #
+    # Current SQL looks something like this:
+    #
+    #  SELECT DISTINCT
+    #     SIMILARITY(name_en, 'steve'),
+    #     SIMILARITY(name_zh_tw, 'steve'),
+    #     organisations.*
+    #  FROM organisations
+    #  WHERE
+    #     SIMILARITY(name_en, 'steve') > 0.1 OR
+    #     SIMILARITY(name_zh_tw, 'steve') > 0.1
+    #  ORDER BY
+    #     SIMILARITY(name_en, 'steve') DESC,
+    #     SIMILARITY(name_zh_tw, 'steve') DESC
+    #
+    def search(search_text)
+      similarities = search_props.map { |f| "SIMILARITY(#{f}, #{sanitize(search_text)})" }
+
+      fields = similarities + ["#{table_name}.*"]
+      conditions = similarities.map { |s| "#{s} > #{similarity_threshold}" }
+      ordering = similarities.map { |s| "#{s} DESC" }
+
+      select(fields.join(','))
+        .where(conditions.join(' OR '))
+        .order(ordering.join(','))
+        .distinct
+    end
+
+    #
+    # Configuration of fuzzy search
+    #
+    # @param [string[]] props array of columns to search against
+    # @param [float] tolerance the tolerance of the search between 0 and 1 \
+    #     1 requires a perfect match and 0 lets any similarity through
+    #
+    # @return [<Type>] <description>
+    #
+    def configure_search(props: [], tolerance: 0.1)
+      @search_props ||= []
+      @search_props = (@search_props << props).flatten
+      @similarity_threshold = tolerance
+    end
+
+    def similarity_threshold
+      @similarity_threshold.present? ? @similarity_threshold : 0.1
```

Comment:
@patrixr , I don't this this method is required, you can set `attr_accessor :similarity_threshold` which will get set in configure search. 

This line is explicitly passing tolerance as `0.1`. Once it is set to an attribute accessor you can use it anywhere in the class, no need of this ternary operator.
https://github.com/crossroads/api.goodcity/pull/870/files#diff-adfd80ef5cd384d3af22bdfc27cb7a76R11

Reply:
Please let me know if i'm missing anything.

Reply:
@abhinav098 
The configure_search call is optional, meaning we have to handle cases where it isn't called. That's what this is about.

@similarity_threshold could be nil, therefore we always account for that scenario by defining a method to wrap it